src/proto/callbacks.proto

```diff
@@ -334,4 +335,36 @@ message Callback {
   message CallbackPrintModelDescription {
   }
 
+  message CallbackSummarizeImages {
+
+    message SelectionStrategy {
+
+      oneof strategy_type {
+        CategoricalAccuracyStrategy categorical_accuracy = 1;
+        AutoencoderStrategy autoencoder = 2;
```

Comment:
I'm not thrilled with these class names since they imply limited use-cases. The functionality is actually very general (I can see them very helpful with GANs), so it would be better to name them based on their behavior rather than the expected use-case. Maybe "SelectWithBool" and "SelectAllAndTagWithSampleIndex".

Reply:
What about `SelectWithBool` and `TrackSampleIDs`? It doesn't track _all_ images, just a fixed number. In a world of consistent sample numbering, we'd like to make it possible for users to say "Track the samples named `abcd1234` and `efgh5678`" But since things are not that nice yet, we just set the number of images to track, and it's the first set of however many that is that gets tracked throughout training.